scenarios.py

```diff
@@ -1,8 +1,108 @@
 from rice import *
-
+import random
+from math import ceil
 _FEATURES = "features"
 _ACTION_MASK = "action_mask"
 
+class CarbonLeakage(Rice):
+
+    """
+    Scenario to test whether carbon leakage occurs.
+
+    Carbon leakage is an increase of emissions in one region as a 
+    result of a policy to decrease emissions in another region
+
+    We can check for carbon leakage at the policy level (do they change their mitigation rate)
+    and at the emissions level (do their absolute emissions increase)
+
+    Followup experiment
+    - create a random club of a given minimum mitigation rate
+    - run the rollout with the club and measure emissions of non-club members and measure mitigation rates of non-club members
+    - reset the env and re-run the env without the club (self.control = True) and measure the same
+    - compare the emissions / mitigation rates of the non-club members in the presence and absence of the club
+    - NOTE: it may be that emissions need to be normalized w.r.t. emissions as carbon
+    """
+
+    def __init__(self,
+                 num_discrete_action_levels=10,  # the number of discrete levels for actions, > 1
+                 negotiation_on=False, # If True then negotiation is on, else off
+                 scenario="CarbonLeakage",
+                 action_space_type="discrete",  # or "continuous"
+                 dmg_function="base",
+                 carbon_model="base",
+                 temperature_calibration="base",
+                 prescribed_emissions=None
+            ):
+        super().__init__(negotiation_on=negotiation_on,  # If True then negotiation is on, else off
+                scenario=scenario,
+                num_discrete_action_levels=num_discrete_action_levels, 
+                action_space_type=action_space_type,  # or "continuous"
+                dmg_function=dmg_function,
+                carbon_model=carbon_model,
+                temperature_calibration=temperature_calibration,
+                prescribed_emissions=prescribed_emissions)
+        
+        #if its the control group, don't apply the club rules
+        
+        self.control = False
+        self.training = True
+        self.minimum_mitigation_rate = 8
+        self.club_size = ceil(self.num_regions/2)
+        self.club_members = random.sample(range(0, self.num_regions + 1), self.club_size)
+
+    def reset(self, *, seed=None, options=None):
+        obs, info = super().reset(seed=seed, options=options)
+
+        #recreate club each time
+        if self.training:
+            self.club_members = random.sample(range(0, self.num_regions + 1), self.club_size)
+
+        #during training, switch up control conditions
+        if self.training:
+            if random.uniform(0,1) < 0.3:
+                self.control = True
```

Comment:
Why there is 0.3 chance of setting control=True? What is the logic behind this? Thx!

Reply:
See pr description. I want that agents are also exposed the rollouts in which there is no club, this way they have an idea how much damages they would expect to receive without others doing the mitigation for them.